examples/provisioner/launchtemplates/windows-custom-userdata.yaml

```diff
@@ -0,0 +1,38 @@
+# This example provisioner will provision instances using the Windows 2022 EKS-Optimized AMI.
+# The UserData defined in spec.UserData needs to be in XML format,
+# and will be prepended to a Karpenter managed section that will bootstrap the kubelet.
+# This example also applies to the Windows 2019 EKS-Optimized AMI.
+# See https://docs.aws.amazon.com/AWSEC2/latest/WindowsGuide/ec2-windows-user-data.html for more details regarding Windows XML userdata
+
+apiVersion: karpenter.sh/v1alpha5
+kind: Provisioner
+metadata:
+  name: windows2022
+spec:
+  requirements:
+    - key: kubernetes.io/os
+      operator: In
+      values: ["windows"]
+  providerRef:
+    name: windows2022
+---
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: windows2022
+  annotations:
+    kubernetes.io/description: "Nodes running Windows Server 2022"
+spec:
+  subnetSelector:
+    karpenter.sh/discovery: "${CLUSTER_NAME}" # replace with your cluster name
+  securityGroupSelector:
+    karpenter.sh/discovery: "${CLUSTER_NAME}" # replace with your cluster name
+  amiFamily: Windows2022
+  metadataOptions:
+    httpProtocolIPv6: disabled
+    httpTokens: required
+  userData: |
+    <powershell>
+    New-Item -Path 'C:\temp\' -ItemType Directory
```

Comment:
I'm wondering if we should show an example that allows for updating kubeletConfiguration if that's possible on the Windows AMI as well?

Reply:
I agree its a good idea to have a more practical use case specified as an example such as updating the kubelet configuration. Unfortunately, at the moment, the bootstrap only supports arguments being passed directly to it as the script is invoked e.g. `.\Start-EKSBootstrap.ps1 -KubeletExtraArgs "--max-pods=110"`.